Semantics.fs

```diff
@@ -131,10 +127,10 @@ type Write =
 /// <param name="value">The value written to the eventual destination.</param>
 /// <param name="map">The write map being extended.</param>
 /// <returns>The extended write map.</returns>
-let markWrite (var : CTyped<Sym<Var>>) (idxPath : IntExpr<Sym<Var>> list)
-  (value : Expr<Sym<Var>>)
-  (map : Map<CTyped<Sym<Var>>, Write>)
-  : Map<CTyped<Sym<Var>>, Write> =
+let markWrite (var : TypedVar) (idxPath : IntExpr<Sym<Var>> list)
+  (value : Expr<Sym<Var>> option)
+  (map : Map<TypedVar, Write>)
+  : Map<TypedVar, Write> =
```

Comment:
Variables can no longer be symbolic (though array indices, and thus general lvalues, can: `foo[%{numeric_symbol}(bar)]` is valid), and assignments can be havocs, hence the changes here.